tests/e2e/multicluster/multicluster_multiprimary_test.go

```diff
@@ -0,0 +1,479 @@
+	"github.com/istio-ecosystem/sail-operator/pkg/test/util/supportedversion"
+	. "github.com/istio-ecosystem/sail-operator/tests/e2e/util/gomega"
+	"github.com/istio-ecosystem/sail-operator/tests/e2e/util/helm"
+	"github.com/istio-ecosystem/sail-operator/tests/e2e/util/istioctl"
+	"github.com/istio-ecosystem/sail-operator/tests/e2e/util/kubectl"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	common "github.com/istio-ecosystem/sail-operator/tests/e2e/util/common"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Multicluster deployment models", Ordered, func() {
+	SetDefaultEventuallyTimeout(180 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	// debugInfoLogged := false
+
+	BeforeAll(func(ctx SpecContext) {
+		if !skipDeploy {
+			// Deploy the Sail Operator on both clusters
+			Expect(kubectl.CreateNamespace(namespace, kubeconfig)).To(Succeed(), "Namespace failed to be created on Primary Cluster")
+			Expect(kubectl.CreateNamespace(namespace, kubeconfig2)).To(Succeed(), "Namespace failed to be created on Remote Cluster")
+
+			Expect(helm.Install("sail-operator", filepath.Join(project.RootDir, "chart"), "--namespace "+namespace, "--set=image="+image, "--kubeconfig "+kubeconfig)).
+				To(Succeed(), "Operator failed to be deployed in Primary Cluster")
+
+			Eventually(common.GetObject).
+				WithArguments(ctx, clPrimary, kube.Key(deploymentName, namespace), &appsv1.Deployment{}).
+				Should(HaveCondition(appsv1.DeploymentAvailable, metav1.ConditionTrue), "Error getting Istio CRD")
+			Success("Operator is deployed in the Primary namespace and Running")
+
+			Expect(helm.Install("sail-operator", filepath.Join(project.RootDir, "chart"), "--namespace "+namespace, "--set=image="+image, "--kubeconfig "+kubeconfig2)).
+				To(Succeed(), "Operator failed to be deployed in Remote Cluster")
+
+			Eventually(common.GetObject).
+				WithArguments(ctx, clRemote, kube.Key(deploymentName, namespace), &appsv1.Deployment{}).
+				Should(HaveCondition(appsv1.DeploymentAvailable, metav1.ConditionTrue), "Error getting Istio CRD")
+			Success("Operator is deployed in the Remote namespace and Running")
+		}
+	})
+
+	Describe("Multi-Primary Multi-Network configuration", func() {
+		// Test the Multi-Primary Multi-Network configuration for each supported Istio version
+		for _, version := range supportedversion.List {
+			Context("Istio version is: "+version.Version, func() {
+				When("Istio resources are created in both clusters with multicluster configuration", func() {
+					BeforeAll(func(ctx SpecContext) {
+						pushIntermediateCA()
+
+						multiclusterYAML := `
+apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s
+  values:
+    global:
+      meshID: %s
+      multiCluster:
+        clusterName: %s
+      network: %s`
```

Comment:
Non blocking question: is there not a generated sail-operator go client we could use for the e2e tests?